### ISSUE TITLE
Allow users to Search ACL Rule

### ISSUE THREAD

**Issue by btzq** [apache/cloudstack#8859]
##### ISSUE TYPE

 * Improvement Request

##### COMPONENT NAME

~~~
Improvement Request (UI, Functionality)
~~~

##### CLOUDSTACK VERSION

~~~
4.19
~~~

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

N/A

##### SUMMARY

Note: This ticket is broken down as requested from the origianl post, https://github.com/apache/cloudstack/issues/8841

**Allow users to Search ACL Rule**

- In large scale projects like ours, we have around 20-40 ACL Rules.
- As the rules increase, its hard to read and identify.
- Supporting search ACL Rule would help.

##### STEPS TO REPRODUCE

N/A

~~~
N/A
~~~

##### EXPECTED RESULTS

~~~
To be able to manage ACL Rules more easily for large scale projects
~~~

##### ACTUAL RESULTS

~~~
Challenging to use ACL rules for large scale projects.
~~~

**Comment by DaanHoogland** [comment]
There is no generic `keyword` search (working!) as for other list APIs.
There is no search on the field `reason`.
There is no search by order/`number`.
There is no search by `cidr`. 
There is no search by ....

for other fields the search is implemented, @btzq : `networkId`, `id`, traffictype`, `protocol`, `action`

**Comment by DaanHoogland** [comment]
@btzq , I started looking at this one, but am a bit strugling with still. I think we can improve a bit on what the extension of functionality would be. Care to chime in?

In #9566 I am adding reason to the generic keyword search listNetworkACLs command, but that is for the lists and not for the items.

Am I correct in understanding that this is mainly a UI issue?

**Comment by btzq** [comment]
Hey @DaanHoogland , just to clarify you mean 'Lists' and 'Items' are as below, right?

<img width="1293" alt="Screenshot 2024-08-23 at 10 35 34 am" src="https://github.com/user-attachments/assets/983978e0-f7e7-48ea-a42c-11acdb5016ee">

If yes, then in my opinion, just being bale to search the specific List/Rule will be sufficient. Its just to make navigating the dozens of List/Rule easier by only showing me the row users wanna see.

**Comment by DaanHoogland** [comment]
I have implemented search for protocol and reason in #9566 @btzq . I think the other items are going to be more difficult but I'll think about those. Can you have a look/test?

**Comment by btzq** [comment]
Im not sure how to compile Cloudstack from source tho... What about a video? I could verify it there.
